uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-k8s/src/main/java/org/uberfire/java/nio/fs/k8s/K8SFileSystemProvider.java

```diff
@@ -0,0 +1,337 @@
+import org.uberfire.java.nio.file.OpenOption;
+import org.uberfire.java.nio.file.Path;
+import org.uberfire.java.nio.file.attribute.BasicFileAttributes;
+import org.uberfire.java.nio.file.attribute.FileAttribute;
+import org.uberfire.java.nio.file.attribute.FileAttributeView;
+import org.uberfire.java.nio.fs.cloud.CloudClientFactory;
+import org.uberfire.java.nio.fs.file.SimpleFileSystemProvider;
+
+import static org.kie.soup.commons.validation.PortablePreconditions.checkCondition;
+import static org.kie.soup.commons.validation.PortablePreconditions.checkNotNull;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemConstants.CFG_MAP_ANNOTATION_FSOBJ_SIZE_KEY;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemConstants.CFG_MAP_FSOBJ_CONTENT_KEY;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemConstants.K8S_FS_SCHEME;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.createOrReplaceFSCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.createOrReplaceParentDirFSCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.deleteAndUpdateParentCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.getFsObjCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.getPathByFsObjCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.isDirectory;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.isRoot;
+
+public class K8SFileSystemProvider extends SimpleFileSystemProvider implements CloudClientFactory {
+    private static final Logger logger = LoggerFactory.getLogger(K8SFileSystemProvider.class);
+    public K8SFileSystemProvider() {
+        super(null, OSType.UNIX_LIKE);
+        this.fileSystem = new K8SFileSystem(this, K8SFileSystem.UNIX_SEPARATOR_STRING);
+    }
+
+    @Override
+    public String getScheme() {
+        return K8S_FS_SCHEME;
+    }
+
+    @Override
+    public InputStream newInputStream(final Path pathIn,
+                                      final OpenOption... options) {
+        checkNotNull("path", pathIn);
+        Path path = toAbsoluteRealPath(pathIn);
+        checkFileNotExistThenThrow(path, false);
+        logger.info("Open InputStream to file [{}]", path);
+        return Channels.newInputStream(new K8SFileChannel(path, this));
+    }
+
+    @Override
+    public OutputStream newOutputStream(final Path pathIn,
+                                        final OpenOption... options) {
+        checkNotNull("path", pathIn);
+        Path path = toAbsoluteRealPath(pathIn);
+        logger.info("Open OutputStream to file [{}]", path);
+        return Channels.newOutputStream(new K8SFileChannel(path, this));
+    }
+
+    @Override
+    public FileChannel newFileChannel(final Path pathIn,
+                                      final Set<? extends OpenOption> options,
+                                      final FileAttribute<?>... attrs) {
+        checkNotNull("path", pathIn);
```

Comment:
I think this is not needed as we throw an exception one line below

Reply:
It's good to have it for helping debugging client code. IMHO

Reply:
Sure, I am 100% for keeping it in place, however if you check `checkNotNull` implementation, you will see you should invoke as `checkNotNull("pathIn", pathIn);` or `checkNotNull("path", path);` to get proper error message. The second option requires method signature update of course.

Reply:
Updated the usage properly to something like: `checkNotNull("path", path);` across the board.